components/popover/stories/popover.stories.js

```diff
@@ -38,147 +47,140 @@ export default {
-		chromatic: { delay: 2000 },
 		docs: {
 			story: {
-				height: "300px"
+				height: "200px",
 			}
 		},
+		// packageJson,
+		// metadata,
 	},
 	decorators: [
 		(Story, context) => html`<div style="padding: 16px">${Story(context)}</div>`
 	],
 };
 
+/**
+ * By default, popovers do not have a tip. Popovers without a tip should be used when the source has a
+ * visually distinct down state, in order to show the connection between the popover and its source.
+ *
+ * This example uses the [menu](?path=/docs/components-menu--docs) component within the popover, and a button as the source.
+ */
+// TODO: reference PopoverGroup() instead of Template()
+// export const Default = PopoverGroup.bind({});
 export const Default = Template.bind({});
-Default.play = async ({ canvasElement }) => {
-	const canvas = within(canvasElement);
-	await userEvent.click(canvas.getByRole("button"));
-};
-Default.args = {};
-
-export const WithTip = Template.bind({});
-WithTip.play = async ({ canvasElement }) => {
-	const canvas = within(canvasElement);
-	await userEvent.click(canvas.getByRole("button"));
-};
-WithTip.args = {
-	withTip: true,
+Default.args = {
+	trigger: (passthroughs, context) => ActionButton({
+		isSelected: true,
+		label: "Actions",
+		...passthroughs,
+	}, context),
+	content: [
+		(passthroughs, context) => Menu({
+			items: [
+				{
+					iconName: "Edit",
+					label: "Edit",
+				},
+				{
+					iconName: "Copy",
+					label: "Copy",
+				},
+				{
+					iconName: "Move",
+					label: "Move",
+				},
+				{
+					iconName: "Delete",
+					label: "Delete",
+				},
+			],
+			...passthroughs,
+		}, context),
+	],
 };
 
+/**
+ * A popover can be nested within another popover.
+ */
 export const Nested = Template.bind({});
-Nested.play = async ({ canvasElement }) => {
-	const canvas = within(canvasElement);
-	await userEvent.click(canvas.getAllByRole("button")[0]);
-	await userEvent.click(canvas.getAllByRole("button")[1]);
-};
+Nested.tags = ["is-hidden-story", "!dev"];
 Nested.args = {
-	nested: true,
-	testId: "popover-nested",
-	id: "popover-nested",
-	triggerId: "trigger-nested",
+	position: "end-top",
 	isOpen: true,
-	customStyles: {
-		"margin-inline-start": "8px",
-	},
-	trigger: (passthroughs) => ActionButton({
-		label: "Hop on pop(over)",
-		id: "trigger-nested",
+	trigger: (passthroughs, context) => ActionButton({
+		label: "Actions",
 		...passthroughs,
-	}),
+	}, context),
 	content: [
-		() => Menu({
+		(passthroughs, context) => Menu({
 			items: [
 				{
 					iconName: "Edit",
 					label: "Edit",
 				},
 			],
-		}),
-		() => Nested({
-			position: "right",
-			testId: "popover-nested-2",
-			id: "popover-nested-2",
-			triggerId: "trigger-nested-2",
+			...passthroughs,
+		}, context),
+		(passthroughs, context) => Template({
+			position: "end-top",
 			isOpen: true,
 			customStyles: {
-				"margin-inline-start": "136px",
-				"margin-block-start": "32px"
+				"margin": "var(--spectrum-popover-nested-container-spacing)",
```

Comment:
This was my attempt at adding the "container padding" that is outlined in Figma. It felt weird to put those styles on the popover itself, i.e.: 
```
& > .spectrum-Popover {
  margin: var(--spectrum-popover-nested-container-spacing);
}
```

And then users can add those styles if they need. Any thoughts on where these styles are supposed to go?

...since it certainly feels like they belong on the containing element. So that's what this is doing- putting that container padding on the popover wrapper. 

https://github.com/adobe/spectrum-css/blob/61af0743f874d799b8fbb789aff8ad2580c5994a/components/popover/stories/template.js#L165

Reply:
Great question! I don't really have an answer, but I did look into it and will share what I've found so far:

Strangely enough, you [asked](https://github.com/adobe/spectrum-css/pull/2743#discussion_r1672743196) about this exact thing when we did S2 for Tooltip, which has the same container padding token in its [spec](https://www.figma.com/design/eoZHKJH9a3LJkHYCGt60Vb/S2-token-specs?node-id=9706-2117&m=dev), and there I think we decided that it was sort of out of our hands to ensure that that token was used in the implementation.

It's also worth mentioning that, if I dig through the [wiki](https://wiki.corp.adobe.com/display/AdobeDesign/Spectrum+Design%3A+Token+Specs) for the S1 specs, I can find the Popover S1 token specs in the XD file from the [Sharepoint](https://adobe.sharepoint.com/sites/Spectrum/Shared%20Documents/Forms/AllItems.aspx?id=%2Fsites%2FSpectrum%2FShared%20Documents%2FSpectrum%20Design%2FToken%20specs), and the container padding doesn't change from S1 to S2, and it doesn't seem like it was implemented anywhere in S1 (it's the same case for tooltip, if you're curious).

I think it's fine if we want to add some margin to ensure that there's spacing here, but I have a couple of follow-up questions about that:
1. Is this `--spectrum-popover-nested-container-spacing` custom prop meant to be used for all container spacing, or is there a reason it's only used for the nested container/only named for nested container spacing?
2. Was the intention to put this margin on the wrapper? It looks like this margin is ending up on the popover (on the same element as `.spectrum-Popover`), not the wrapper, but applying them to `popoverWrapperStyles` instead puts the margin on the action button, which isn't what we want either. It seems like we'd probably want some space here anyway (maybe?) but I'm not sure we need to use the token (but if I'm missing something let me know!).
    ![image](https://github.com/user-attachments/assets/2f492c22-0a90-4633-8a50-d7bb5f5c0143)
    ![image](https://github.com/user-attachments/assets/6d592b1f-8dd2-4b77-a7a1-7bc410166b18)
3. As far as I can tell, the `--spectrum-popover-nested-container-spacing` is defined in index.css then not used in index.css, is that going to cause some linting issues in `main`? (I would have thought there would be some in this branch too, but I didn't see any.)

Does any of that help here?

Reply:
I am totally fine with _not_ implementing the container padding, but I suppose the question is the same for this specific, nested popover story.

> Is this --spectrum-popover-nested-container-spacing custom prop meant to be used for all container spacing, or is there a reason it's only used for the nested container/only named for nested container spacing?

I don't believe this should be used for all container spacing. I'm using it only for the nested popover because those 2 popovers "touch" now. The spacing between the trigger and the popover itself is dictated by the `animation-distance`. That animation distance if from the trigger, which in this case is 8px from the action buttons. But now with the new, additional padding popover has (which is also 8px), the `.spectrum-Popover` divs "touch." 

I suppose instead of `spacing-100`, I could use the `animation-distance` token?

without `--spectrum-popover-nested-container-spacing`:
<img width="331" alt="Screenshot 2024-11-18 at 1 39 27 PM" src="https://github.com/user-attachments/assets/3f187f81-3d80-4d5c-9b72-1243457abcaa">

with `--spectrum-popover-nested-container-spacing`:
<img width="347" alt="Screenshot 2024-11-18 at 1 40 31 PM" src="https://github.com/user-attachments/assets/a8fa1e74-ad7c-48f6-9d25-bc2ca38288d5">

> As far as I can tell, the --spectrum-popover-nested-container-spacing is defined in index.css then not used in index.css, is that going to cause some linting issues in main? (I would have thought there would be some in this branch too, but I didn't see any.)

Correct, I didn't actually use it in the CSS. I only used it in the stories file, so I moved it and just defined it in the stories file too. Good call.

Let me know if this solution also addresses your second question about not needing to use that token! https://github.com/adobe/spectrum-css/pull/3365/commits/3ee1873a2fd21f2f69fc93c0e6f055bb4116151c

Reply:
This looks great with the animation distance token!

Reply:
How's this? https://github.com/adobe/spectrum-css/pull/3365/commits/a0ff0b26c9651db3d7f7d7cd2e2339eb228e854c